### PR TITLE
Add CloudWatchLogsDiscovery

### PR DESCRIPTION
**Pull request by stefansjfw**
\+ some auto formatting files

Needed for AWS benchmark rules 3.*

### CODE REVIEW

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/CloudWatchLogsDiscovery.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  private static final String SERVICE = "cloudWatchLogs";
+
+  @Override
+  public String service() {
+    return SERVICE;
+  }
+
+  @Override
+  public List<Region> getSupportedRegions() {
+    return CloudWatchClient.serviceMetadata().regions();
```
Comment (PrzemyslawTusinski) [inline magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/CloudWatchLogsDiscovery.java:46]: CloudWatchLogsClient

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/CloudWatchLogsDiscovery.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    discoverLogs(mapper, session, region, emitter, client, account);
+  }
+
+  private void discoverLogs(ObjectMapper mapper, Session session, Region region, Emitter emitter, CloudWatchLogsClient client, String account) {
+    final String RESOURCE_TYPE = "AWS::CloudWatchLogs::MetricFilter";
+
+    try {
+      client.describeMetricFiltersPaginator().metricFilters().forEach(metricFilter -> {
+        var data = new AWSResource(metricFilter.toBuilder(), region.toString(), account, mapper);
+        data.arn = String.format("arn:aws:cloudwatchlogs:%s:%s:metric-filter/%s", region, account, metricFilter.filterName());
```
Comment (PrzemyslawTusinski) [inline magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/CloudWatchLogsDiscovery.java:62]: Is this proper arn? Can you provide a link describing it?

Comment (PrzemyslawTusinski) [inline magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/CloudWatchLogsDiscovery.java:62]: Its probably better to discover log groups, then put metric filter for given log group to supplementary configuration

**Comment by PrzemyslawTusinski** [comment]
It's better to merge this to cpsm

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/CloudWatchLogsDiscovery.java (hunk 2 above)
Comment (PrzemyslawTusinski) [reply magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/CloudWatchLogsDiscovery.java:62]: On the second thought no, it's takes forever to discover all of them

Comment (stefansjfw) [reply magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/CloudWatchLogsDiscovery.java:62]: Hm. not sure. I can't find examples of metric-filter arns. I'll check again

Comment (stefansjfw) [reply magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/CloudWatchLogsDiscovery.java:61]: Can't find any occurrence of metric filter arn, looks like it has only name. Did you notice any other resource without arn while working on `discovery`?

Comment (PrzemyslawTusinski) [reply magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/CloudWatchLogsDiscovery.java:61]: Every discovered resource has arn so far, if I remember correctly.
I've checked in console and metric-filters doesn't have arn at all.

Maybe discovering log groups which has arn and putting metric filters in supplementary configuration is better option.
I've checked and it doesn't take as much time as I initially thought. About 10m on my machine

Comment (stefansjfw) [reply magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/CloudWatchLogsDiscovery.java:61]: ok. Thanks for checking. I'll see if that's better approach

Comment (stefansjfw) [reply magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/CloudWatchLogsDiscovery.java:61]: Hm. I'm not sure discovering of few thousand log groups to get just a few metric filters is worth that time. Anyway, metric filters are just side result to get the final one - so far. I would prefer leave it this way, at least until metric filters and/or log groups are needed as final result of some query

**Review by belosh59 — APPROVED** [review]

**Review by PrzemyslawTusinski — APPROVED** [review]

**Pull request merged**